framework/workspace/view/workspacelistmodel.cpp

```diff
@@ -0,0 +1,210 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "workspacelistmodel.h"
+
+#include "log.h"
+
+using namespace mu::workspace;
+
+static const QString NAME_KEY("name");
+static const QString IS_SELECTED_KEY("isSelected");
+static const QString IS_REMOVABLE_KEY("isRemovable");
+static const QString INDEX_KEY("index");
+
+WorkspaceListModel::WorkspaceListModel(QObject* parent)
+    : QAbstractListModel(parent)
+{
+}
+
+void WorkspaceListModel::load()
+{
+    beginResetModel();
+    m_workspaces.clear();
+
+    RetVal<IWorkspacePtrList> workspaces = workspacesManager()->workspaces();
+    if (!workspaces.ret) {
+        LOGE() << workspaces.ret.toString();
+    }
+
+    IWorkspacePtr currentWorkspace = workspacesManager()->currentWorkspace().val;
+    IWorkspacePtr selectedWorkspace;
+
+    for (const IWorkspacePtr& workspace : workspaces.val) {
+        if (workspace == currentWorkspace) {
+            selectedWorkspace = workspace;
+        }
+
+        m_workspaces << workspace;
+    }
+
+    std::sort(m_workspaces.begin(), m_workspaces.end(), [this](const IWorkspacePtr& workspace1, const IWorkspacePtr& workspace2) {
+        return workspace1->name() < workspace2->name();
```

Comment:
what about the default workspace? should it be at the beginning of the list?

Reply:
I don't know. Probably I will implement better sorting later, this one is only for demonstration